Support JSON::XS 4

ppisar opened the pull request:
JSON::XS 4 has the same API as version 3. It only enables allow_nonref
by default.

CPAN RT#127753

ppisar committed:
Support JSON::XS 4

JSON::XS 4 has the same API as version 3. It only enables allow_nonref
by default.

CPAN RT#127753

karenetheridge commented:
I don't think this is a good change. We should not let the underlying backend decide what the default is for this option, because the user won't know what backend+version he is using, so depending on what is installed he will get different results.

We will have to explicitly pass `allow_nonref => 0` now if the user didn't provide it.

I will be looking at applying patches for JSON::Any and JSON::MaybeXS in the near future. (and we also need tests for this in `xt/author`.)

ppisar committed:
Disable allow_nonref in JSON::XS 4 and JSON:PP 3

This patch disables allow_nonref feature in JSON::XS >= 4 and JSON::PP >= 3.
The reason is to provide a uniform behavior including backends that
do not support it.

<https://github.com/karenetheridge/JSON-Any/pull/2>

Bluewind commented:
Given perl 5.30 has been released and it includes JSON::XS 4.0, what's the status of this?
